File: app/backend/internal/models/notification.go
package models

import "time"

// Notification represents a system notification for a user
type Notification struct {
	ID            uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID        uint      `gorm:"not null;index:idx_user_read" json:"user_id"`
	User          User      `gorm:"foreignKey:UserID;constraint:OnDelete:CASCADE;" json:"user"`
	Title         string    `gorm:"type:varchar(255);not null" json:"title"`
	Message       string    `gorm:"type:text;not null" json:"message"`
	Type          *string   `gorm:"type:varchar(50);default:'info'" json:"type"`
	IsRead        *bool     `gorm:"default:false;index:idx_user_read" json:"is_read"`
	RelatedEntity *string   `gorm:"type:varchar(50)" json:"related_entity"`
	RelatedID     *uint     `json:"related_id"`
	CreatedAt     time.Time `gorm:"autoCreateTime" json:"created_at"`
}

File: app/frontend/src/types/notification.ts
export interface NotificationItem {
  id: number;
  user_id: number;
  title: string;
  message: string;
  type: 'info' | 'success' | 'warning' | 'error';
  is_read: boolean;
  related_entity?: string;
  related_id?: number;
  created_at: string;
}

export interface NotificationListResponse {
  data: NotificationItem[];
  unread_count: number;
}
